### PR TITLE
Bbb/2.13

### PR DESCRIPTION
**Pull request by anadahz**

### CODE REVIEW

**Comment by anadahz** [comment]
This patch is required to our installation, upstream watch:
https://github.com/n0emis/ansible-role-bigbluebutton/pull/74

**Comment by anadahz** [comment]
https://travis-ci.org/github/PrivacyLx/devops/jobs/763911247
Ready to merge.

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ lint: |
 platforms:
   - name: ubuntu1604-instance
     box: generic/ubuntu1604
```
Comment (francisco-core) [inline ansible/roles/bigbluebutton/molecule/default/molecule.yml:14]: Not important for now, but just a note that: (emphasis mine; taken from [here](https://ubuntu.com/16-04))

> Ubuntu 16.04 LTS is supported until 2024 through Canonical’s Extended Security Maintenance (ESM) product. Xenial enters the ESM period in **April 2021**, with security patches provided for an additional three years beyond the traditional five-year standard support.

Related BBB issue: https://github.com/bigbluebutton/bigbluebutton/issues/8474#issuecomment-802798671


Created an issue for it: https://github.com/PrivacyLx/privacylx-issue-tracker/issues/97

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ lint: |
...
+    box_version: 3.0.1
```
Comment (francisco-core) [inline ansible/roles/bigbluebutton/molecule/default/molecule.yml:15]: Why is this needed and where did the number come from?

I was only able to find these versions: https://app.vagrantup.com/ubuntu/boxes/xenial64

File: ansible/roles/bigbluebutton/tasks/main.yml [hunk 3]
```diff
@@ -33,12 +33,3 @@
...
   delegate_to: localhost
-
-  # https://docs.bigbluebutton.org/greenlight/gl-admin.html#creating-an-administrator-account
-- name: Create Greenlight admin user
-  # https://yamllint.readthedocs.io/en/stable/disable_with_comments.html
-  # yamllint disable rule:line-length
-  command: >
-    docker exec greenlight-v2 bundle exec rake
-    user:create['{{ bbb_adm.name }}','{{ bbb_adm.email }}','{{ bbb_adm.pass }}','admin']
-  # yamllint enable rule:line-length
```
Comment (francisco-core) [inline ansible/roles/bigbluebutton/tasks/main.yml:44]: Ok, we remove this from here and then add the user information to the variable `bbb_greenlight_users`. But that variable is never used in a task. So I think this means that we are not actually creating the default admin user, which I think is a bug.

Don't we need to add some task to create that for us? Or did I miss something?

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml [hunk 4]
```diff
@@ -27,7 +28,6 @@ provisioner:
     links:
       group_vars: ../../../../inventory/molecule/group_vars/
   playbooks:
-    prepare: prepare.yml
```
Comment (francisco-core) [inline ansible/roles/bigbluebutton/molecule/default/molecule.yml:30]: The `prepare.yml` is still in the commit. So we either remove it or we don't remove the above line. But not both.

File: ansible/roles/bigbluebutton/molecule/default/prepare.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+---
```
Comment (francisco-core) [inline ansible/roles/bigbluebutton/molecule/default/prepare.yml:1]: This file is no longer referenced in `molecule.yaml` so it's possibly not needed.

File: ansible/roles/update/templates/52unattended-upgrades.j2 [hunk 6]
```diff
@@ -2,13 +2,17 @@
 
 // Controls which packages are upgraded
 Unattended-Upgrade::Allowed-Origins {
+{% if 'bigbluebutton' in group_names %}
+        "${distro_id}:${distro_codename}-security";
+{% else %}
         "${distro_id}:${distro_codename}";
         "${distro_id}:${distro_codename}-updates";
         "${distro_id}:${distro_codename}-security";
+{% endif %}
```
Comment (francisco-core) [inline ansible/roles/update/templates/52unattended-upgrades.j2:11]: Why do we need this `if`? Why not just the line `${distro_id}:${distro_codename}-security";`?

The whole point of this role is to install BBB. So we might only say that it needs the security updates.

File: ansible/roles/bigbluebutton/defaults/main.yml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 galaxy_path: "{{ playbook_dir }}/roles.galaxy"
 # Git commit or release version of bbb_role_git
-bbb_role_ver: 5f1e22498921e10f2c18bfb4a311aede18a2b62f
+bbb_role_ver: 77697c9f5fd279f748324dfb9569e2ee5e0daa6e
```
[the remark this answers is not in the document]
Comment (anadahz) [reply ansible/roles/bigbluebutton/defaults/main.yml]: This is the latest git commit.

File: ansible/roles/bigbluebutton/defaults/main.yml [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 galaxy_path: "{{ playbook_dir }}/roles.galaxy"
 # Git commit or release version of bbb_role_git
-bbb_role_ver: 5f1e22498921e10f2c18bfb4a311aede18a2b62f
+bbb_role_ver: 0104e9eefcaf88b9129a287953984fcaa63adc41
```
[the remark this answers is not in the document]
Comment (anadahz) [reply ansible/roles/bigbluebutton/defaults/main.yml:5]: This is the latest git commit.

[the remark this answers is not in the document]
Comment (anadahz) [reply ansible/roles/bigbluebutton/defaults/main.yml:5]: Yes, this is actually the [latest git commit](https://github.com/n0emis/ansible-role-bigbluebutton/commit/0104e9eefcaf88b9129a287953984fcaa63adc41).

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml (hunk 1 above)
Comment (anadahz) [reply ansible/roles/bigbluebutton/molecule/default/molecule.yml:14]: Thanks for opening an issue.

File: ansible/roles/bigbluebutton/defaults/main.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (francisco-core) [reply ansible/roles/bigbluebutton/defaults/main.yml]: Ah. I see. The docker ones are not in perfect sync with the git repo.

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml (hunk 2 above)
Comment (anadahz) [reply ansible/roles/bigbluebutton/molecule/default/molecule.yml]: Ah good catch this is not relevant any more as I was testing different versions. I fixed this in: https://github.com/PrivacyLx/devops/pull/55/commits/c559cdfc86214e921675a492bf48552cb8024fc4

For reference this is the image that we are using:
https://app.vagrantup.com/generic/boxes/ubuntu1604

File: ansible/roles/bigbluebutton/defaults/main.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (francisco-core) [reply ansible/roles/bigbluebutton/defaults/main.yml]: Don't know where I saw the other one. Sorry. Confusion.

File: ansible/roles/bigbluebutton/defaults/main.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (francisco-core) [reply ansible/roles/bigbluebutton/defaults/main.yml:5]: Don't know where I saw the other one. Sorry. Confusion.

**Review by francisco-core — COMMENTED** [review]
replied

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml (hunk 4 above)
Comment (anadahz) [reply ansible/roles/bigbluebutton/molecule/default/molecule.yml]: I missed that!
The playbook in `prepare.yml` is required, as you can see in the the CI log it is running: https://travis-ci.org/github/PrivacyLx/devops/jobs/763911247#L594

I have added it again in: https://github.com/PrivacyLx/devops/pull/55/commits/5614b609c20a3a38a78098f994ac46f2f7a44840

File: ansible/roles/bigbluebutton/molecule/default/prepare.yml (hunk 5 above)
Comment (anadahz) [reply ansible/roles/bigbluebutton/molecule/default/prepare.yml:1]: Please see comment: https://github.com/PrivacyLx/devops/pull/55/files#r598654577

File: ansible/roles/bigbluebutton/tasks/main.yml (hunk 3 above)
Comment (anadahz) [reply ansible/roles/bigbluebutton/tasks/main.yml:44]: It's being used by the [upstream's ansible role](https://github.com/n0emis/ansible-role-bigbluebutton).
The variable is set in our inventory here: https://github.com/PrivacyLx/devops/pull/55/files#diff-1244efcc21eb365c7e47066a99ee7979c7e5eb3535ea5faab967d6b6b4cba215R69

File: ansible/roles/update/templates/52unattended-upgrades.j2 (hunk 6 above)
Comment (anadahz) [reply ansible/roles/update/templates/52unattended-upgrades.j2:11]: This template is being used by all other hosts also.
In the case of bigbluebutton host group we only want to trigger unattended upgrades of security packages and not all other packages (as we do other host groups).

The reason of doing this is because of BBB's "sensitivity" with package upgrades. ;)

I hope that this is clear now?

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml (hunk 1 above)
Comment (anadahz) [reply ansible/roles/bigbluebutton/molecule/default/molecule.yml]: Thanks!

File: ansible/roles/bigbluebutton/tasks/main.yml (hunk 3 above)
Comment (francisco-core) [reply ansible/roles/bigbluebutton/tasks/main.yml:44]: I see. Thanks for the clarification.

File: ansible/roles/bigbluebutton/molecule/default/molecule.yml (hunk 4 above)
Comment (francisco-core) [reply ansible/roles/bigbluebutton/molecule/default/molecule.yml]: perfect.

**Comment by anadahz** [comment]
@francisco-core Thank you, for reviewing the PR.
I added some trivial changes, that you catched during the review, thanks again.

File: ansible/roles/update/templates/52unattended-upgrades.j2 (hunk 6 above)
Comment (francisco-core) [reply ansible/roles/update/templates/52unattended-upgrades.j2:11]: ok. makes sense

**Review by francisco-core — APPROVED** [review]
@anadahz ready to merge
